src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests5.cs

```diff
@@ -680,59 +1346,401 @@ class C {
+                Diagnostic(ErrorCode.ERR_IdentifierExpected, "Y.Z").WithLocation(6, 32)
+                );
+        }
+
+        [Fact]
+        public void ExtendedPropertyPatterns_ObsoleteProperty()
+        {
+            var program = @"
+using System;
+class C
+{
+    public void M1(C1 c1)
+    {
+        _ = c1 is { Prop1.Prop2: null };
+    }
+}
+
+class C1
+{
+    [ObsoleteAttribute(""error Prop1"", true)]
+    public C2 Prop1 { get; set; }
+}
+class C2
+{
+    [ObsoleteAttribute(""error Prop2"", true)]
+    public object Prop2 { get; set; }
+}
+";
+            var compilation = CreateCompilation(program, parseOptions: TestOptions.RegularWithExtendedPropertyPatterns);
+            compilation.VerifyEmitDiagnostics(
+                // (7,21): error CS0619: 'C1.Prop1' is obsolete: 'error Prop1'
+                //         _ = c1 is { Prop1.Prop2: null };
+                Diagnostic(ErrorCode.ERR_DeprecatedSymbolStr, "Prop1").WithArguments("C1.Prop1", "error Prop1").WithLocation(7, 21),
+                // (7,27): error CS0619: 'C2.Prop2' is obsolete: 'error Prop2'
+                //         _ = c1 is { Prop1.Prop2: null };
+                Diagnostic(ErrorCode.ERR_DeprecatedSymbolStr, "Prop2").WithArguments("C2.Prop2", "error Prop2").WithLocation(7, 27)
+                );
+        }
+
+        [Fact]
+        public void ExtendedPropertyPatterns_ObsoleteAccessor()
+        {
+            var program = @"
+using System;
+class C
+{
+    public void M1(C1 c1)
+    {
+        _ = c1 is { Prop1.Prop2: null };
+    }
+}
+
+class C1
+{
+    public C2 Prop1
+    {
+        [ObsoleteAttribute(""error Prop1"", true)]
+        get;
+        set;
+    }
+}
+class C2
+{
+    public object Prop2
+    {
+        get;
+        [ObsoleteAttribute(""error Prop2"", true)]
+        set;
+    }
+}
+";
+            var compilation = CreateCompilation(program, parseOptions: TestOptions.RegularWithExtendedPropertyPatterns);
+            compilation.VerifyEmitDiagnostics(
+                // (7,21): error CS0619: 'C1.Prop1.get' is obsolete: 'error Prop1'
+                //         _ = c1 is { Prop1.Prop2: null };
+                Diagnostic(ErrorCode.ERR_DeprecatedSymbolStr, "Prop1.Prop2").WithArguments("C1.Prop1.get", "error Prop1").WithLocation(7, 21)
+                );
+        }
+
+        [Fact]
+        public void ExtendedPropertyPatterns_InaccessibleProperty()
+        {
+            var program = @"
+using System;
+class C
+{
+    public void M1(C1 c1)
+    {
+        _ = c1 is { Prop1.Prop2: null };
+    }
+}
+
+class C1
+{
+    private C2 Prop1 { get; set; }
+}
+class C2
+{
+    private object Prop2 { get; set; }
+}
+";
+            var compilation = CreateCompilation(program, parseOptions: TestOptions.RegularWithExtendedPropertyPatterns);
+            compilation.VerifyEmitDiagnostics(
+                // (7,21): error CS0122: 'C1.Prop1' is inaccessible due to its protection level
+                //         _ = c1 is { Prop1.Prop2: null };
```

Comment:
>Prop2 [](http://example.com/codeflow?start=45&length=5)

Should we report an error on `Prop2` as well?

Reply:
That is not part of this PR, and because the behavior matches pre-existing member access behavior, I'll leave as-is.
I don't think this is simple to fix. `BindInstanceMemberAccess` returns with a `null` `ExpressionSymbol` in this scenario, so when we get to `Prop2` we're only looking at an error type...